src/compile/data/assemble.ts

```diff
@@ -47,7 +47,7 @@ function removeUnnecessaryNodes(node: DataFlowNode) {
  */
 function cloneSubtree(facet: FacetNode) {
   function clone(node: DataFlowNode): DataFlowNode[] {
-    if (!(node instanceof OrderNode)) {  // we can ignore order nodes beacuse they don't change the scale domain
+    if (!(node instanceof OrderNode || node instanceof FacetNode)) {  // we can ignore order nodes beacuse they don't change the scale domain
```

Comment:
> // we can ignore order nodes beacuse they don't change the scale domain

1.  Typo "beacuse" => "because"
2. This is not always true if the scale domain does not have additional sorting.